browser/ios/Plugin/BrowserPlugin.swift

```diff
@@ -0,0 +1,69 @@
+import Foundation
+import Capacitor
+
+@objc(CAPBrowserPlugin)
+public class CAPBrowserPlugin: CAPPlugin {
+    private let implementation = Browser()
+
+    @objc func open(_ call: CAPPluginCall) {
+        // validate the URL
```

Comment:
I would argue not; it's partly a separation of concerns and partly to avoid duplication. This checks if the user passed a valid URL, not what kind of URL it is. The specific protocols that are supported are an internal detail of the implementation. And it makes sense for the implementation to keep that check in case it ever gets used by some other code.